leap_year.py

```diff
@@ -0,0 +1,34 @@
+class LeapYear:
+    def __init__(self, year_input: int):
+        self._year_input = year_input
+
+    def is_year_input_century_year(self) -> bool:
+        return self._year_input % 100 == 0
+
+    def is_century_year_equal_leap_year(self) -> bool:
+        return self._year_input % 400 == 0
+
+    def is_normal_year_equal_leap_year(self) -> bool:
+        return self._year_input % 4 == 0
+
+
+class CliHandler:
+
+    def start(self):
+        while True:
+            try:
+                year_input = int(input('Enter the year:'))
+            except ValueError:
+                print("Invalid input.\tTry again.")
+                continue
+            leap_year = LeapYear(year_input)
+            if leap_year.is_year_input_century_year():
```

Comment:
Just create a single method in the above class which determines the leap year or not and returns a boolean value.